server/src/modules/task/task.interface.ts

```diff
@@ -0,0 +1,8 @@
+export interface TaskDTO {
+  id: number;
+  name: string;
+  mentee_id: number;
+  task_description: string;
+  task_status: string;
+  due_date: Date;
```

Comment:
Does this work? I think DTO objects are always supposed to represent what you're sending or receiving to/from the server, and the NestJS server can't send a JavaScript `Date` object. You might want to make a `due_date_timestamp` field in the DTO that represents a timestamp received from the server (or sent from the client to the server), and then create methods like `serializeTask(task: Task): TaskDTO` and `deserializeTask(taskDTO: TaskDTO): Task` to convert between the two types (e.g. converting the `Date` to a timestamp and the other way).